Feature/snowflake e2e test

zhangchi1 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. Ensure that your code follows our code conventions: https://github.com/feast-dev/feast/blob/master/docs/contributing.md#code-conventions
2. Run unit tests and ensure that they are passing: https://github.com/feast-dev/feast/blob/master/docs/contributing.md#running-unit-tests
3. If your change introduces any API changes, make sure to update the integration tests scripts here: https://github.com/feast-dev/feast/tree/master/tests/e2e
4. Make sure documentation is updated for your PR!
5. Make sure you have signed the CLA https://cla.developers.google.com/clas

-->

**What this PR does / why we need it**:
1. Added integration tests for snowflake batch store in serving 

**Next Step**
1. Mocking snowflake database 

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information about release notes, see kubernetes' guide here:
http://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

zhangchi1 committed:
added retriever tests

e2e test for snowflake sink

updated test cases

finished test cases and updated timezone

fixed timezone mismatch bug

renamed IT file

updated kafka topic constant

swatiarora0208 commented on serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java:
```diff
@@ -0,0 +1,208 @@
+        new KafkaTemplate<>(
+            new DefaultKafkaProducerFactory<>(
+                props, new StringSerializer(), new KafkaSerialization.ProtoSerializer<>()));
+    t.setDefaultTopic("feast-features");
```
topic is used as multiple place .. could be defined as constant

zhangchi1 replied:
Done

zhangchi1 committed:
mocking test step 1: retriever tests in storage

mocking test step 1: retriever tests in storage fixed lint

mocking test step 1: retriever tests in storage fixed lint

zhangchi1 committed:
mocking test step 2: add role config in the sink&write

mocking test step 3: updated jdbc sink and implemented preparewrite and write tests

zhangchi1 commented on storage/connectors/jdbc/pom.xml:
```diff
@@ -146,6 +162,12 @@
+        </dependency>
     </dependencies>
     <properties>
         <maven.test.skip>true</maven.test.skip>
```
I think we should un-skip the new mocking tests. 
And skip those manual tests, assume we still want to include those manual e2e tests in our snowflake branch.

zhangchi1 committed:
mocking test step 3: updated jdbc sink and imp preparewrite, write tests

jmelinav commented on infra/docker/core/Dockerfile.dev:
```diff
@@ -1,6 +1,6 @@
 FROM openjdk:11-jre
 ARG REVISION=dev
-ADD $PWD/core/target/feast-core-$REVISION.jar /opt/feast/feast-core.jar
+ADD $PWD/core/target/feast-core-$REVISION-exec.jar /opt/feast/feast-core.jar
```
this should already be there in master, I pushed it last week.

jmelinav commented on infra/docker/serving/Dockerfile.dev:
```diff
@@ -7,7 +7,7 @@ ARG REVISION=dev
          -O /usr/bin/grpc-health-probe && \
     chmod +x /usr/bin/grpc-health-probe
-ADD $PWD/serving/target/feast-serving-$REVISION.jar /opt/feast/feast-serving.jar
+ADD $PWD/serving/target/feast-serving-$REVISION-exec.jar /opt/feast/feast-serving.jar
```
same as above.

zhangchi1 commented on serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java:
```diff
@@ -0,0 +1,209 @@
+@ActiveProfiles("it")
+@SpringBootTest()
+@Testcontainers
+public class ServingServiceSnowflakeStorageIT {
```
Disable this manual Integration test

zhangchi1 replied:
Add a comment for using customize Feast Core

jmelinav commented on serving/pom.xml:
```diff
@@ -340,6 +346,11 @@
       <scope>test</scope>
     </dependency>
+     <!-- Kafka -->
+    <dependency>
```
if it is used only for test add the scope as test.

zhangchi1 commented on storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java:
```diff
@@ -40,7 +40,7 @@
 
   private Map<String, String> snowflakeConfig = new HashMap<>();
-  private String SFUrl = "jdbc:snowflake://ry42518.us-east-2.aws.snowflakecomputing.com";
+  private String SFUrl = "jdbc:snowflake://nx46274.us-east-2.aws.snowflakecomputing.com";
```
remove all personal "real" url variables

jmelinav commented on serving/src/test/java/feast/serving/it/SnowflakeTestUtils.java:
```diff
@@ -0,0 +1,299 @@
+import java.util.zip.GZIPInputStream;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class SnowflakeTestUtils {
```
merge it with AuthTestUtils.java and rename file.

zhangchi1 commented on serving/src/main/resources/application.yml:
```diff
@@ -78,6 +78,24 @@ feast:
+        role: ACCOUNTADMIN
+        table_name: FEAST_FEATURES_TEST
+        url: jdbc:snowflake://test.snowflakecomputing.com
+        username: username
+        password: password
```
name those configs as fake-configs

jmelinav commented on serving/src/test/resources/application-it.yml:
```diff
@@ -0,0 +1,26 @@
+feast:
+
+  # Indicates the active store. Only a single store in the last can be active at one time. In the future this key
+  # will be deprecated in order to allow multiple stores to be served from a single serving instance
```
there is already application-it.properties. put the defaults there.

jmelinav commented on serving/src/test/resources/docker-compose/docker-compose-it-core.yml:
```diff
@@ -2,8 +2,8 @@ version: '3'
   core:
-    image: gcr.io/kf-feast/feast-core:latest
-    volumes: 
+    image: feast-core:dev1
```
undo this.

zhangchi1 committed:
mocking test step 4: amend tests based on PR comments

zhangchi1 committed:
mocking test step 5: undo dockerfile.dev

zhangchi1 committed:
mocking test step 6: skip IT tests in serving pom
